build.rs

```diff
@@ -0,0 +1,161 @@
+extern crate avr_mcu;
+
+use avr_mcu::*;
+use std::fs::{self, File};
+use std::io;
+use std::io::prelude::*;
+use std::path::{Path, PathBuf};
+
+fn cores_path() -> PathBuf {
+    Path::new(env!("CARGO_MANIFEST_DIR")).join("src").join("cores")
+}
+
+fn core_module_name(mcu: &Mcu) -> String {
+    mcu.device.name.to_lowercase().to_owned()
+}
+
+fn main() {
+    if !cores_path().exists() {
+        fs::create_dir_all(&cores_path()).expect("could not create cores directory");
+    }
+
+    let current_mcu = avr_mcu::current::mcu()
+        .expect("no target cpu specified");
+    generate_cores(&[current_mcu]).unwrap()
+}
+
+fn generate_cores(mcus: &[Mcu]) -> Result<(), io::Error> {
+    for mcu in mcus {
+        generate_core_module(mcu).expect("failed to generate mcu core");
```

Comment:
This returns a result; do we just want to propagate?

Reply:
Or perhaps not until we have an error type to keep the extra information from the `expect` text